studio/components/ui/Charts/BarChart/BarChart.tsx

```diff
@@ -0,0 +1,272 @@
+import { IconBarChart2, Loading } from '@supabase/ui'
+import { useState } from 'react'
+import {
+  BarChart as RechartBarChart,
+  Bar,
+  XAxis,
+  YAxis,
+  Tooltip,
+  Cell,
+  ResponsiveContainer,
+} from 'recharts'
+
+import dayjs from 'dayjs'
+import { formatBytes } from 'lib/helpers'
+import utc from 'dayjs/plugin/utc'
+import { CHART_COLORS } from '../Charts.constants'
+import { BarChartProps, DataPoint, HeaderType } from './BarChart.types'
+import { CategoricalChartState } from 'recharts/types/chart/generateCategoricalChart'
+import { isNumber } from 'lodash'
+
+function dataCheck(data: DataPoint[] | undefined, attribute: string) {
+  const hasData = data?.find(record => record[attribute])
+  return hasData ? true : false
+}
+
+const CustomTooltip = () => null
+const DATE_FORMAT__DATE_ONLY = 'MMM D, YYYY'
+const DATE_FORMAT__WITH_TIME = `${DATE_FORMAT__DATE_ONLY}, hh:mma`
+
+const Header = ({
+  attribute,
+  focus,
+  format,
+  highlightedValue,
+  data,
+  customDateFormat,
+  label,
+  minimalHeader = false,
+  displayDateInUtc = false,
+}: HeaderType) => {
+  const FOCUS_FORMAT = customDateFormat
+    ? customDateFormat
+    : format == '%'
+    ? DATE_FORMAT__WITH_TIME
+    : DATE_FORMAT__DATE_ONLY
+
+  let title = ''
+
+  if (focus) {
```

Comment:
this entire block should not be here. There should not be any parent-specific logic for this common component, since it is an abstraction.

Reply:
This logic should be moved to wherever this is relevant (Reports section).